### PR TITLE
Add relative-path support back.

### PR DESCRIPTION
**Pull request by Kashomon**
This PR adds relative path support back. You can now specify relative
paths for any of the file URLs. File-paths will be rewritten only if the
child-file path (e.g., an object in a component) is a relative path.

Fixes: #227

### CODE REVIEW

File: pkg/build/path_rewriter.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// RelativePathRewriter rewrites paths when the paths are relative paths.
+type RelativePathRewriter struct{}
+
+// RewritePath rewrites file paths if the path is relative, from the
+// original object path, to an object path that's based on a parent component
+// path.
+//
+// For example, if the component path is foo/bar/biff.yaml and the object path
+// is zed/fred.yaml, the object will be rewritten as foo/bar/zed/fred.yaml
+func (rw *RelativePathRewriter) RewriteURL(parent, obj *url.URL) *url.URL {
```
Comment (mikedanese) [inline pkg/build/path_rewriter.go]: Myabe call this Abs() or MakeAbsolute()

File: pkg/build/path_rewriter.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package build
+
+import (
+	"net/url"
+	"path/filepath"
+)
+
+// RelativePathRewriter rewrites paths when the paths are relative paths.
+type RelativePathRewriter struct{}
```
Comment (mikedanese) [inline pkg/build/path_rewriter.go]: Are you thinking about adding any other rewriters? What is the motivation for the struct vs a function?

File: pkg/build/path_rewriter.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		// Only file schemes are supported.
+		return obj
+	}
+	if filepath.IsAbs(obj.Path) {
+		return obj
+	}
+	return &url.URL{
+		Scheme: parent.Scheme,
+		Host:   parent.Host,
+		Path:   filepath.Join(filepath.Dir(parent.Path), obj.Path),
```
Comment (mikedanese) [inline pkg/build/path_rewriter.go]: Do you want to filepath.Clean this as well? Do you want to add any limits like disallowing traversal out of the workspace?

File: examples/cluster/etcd/etcd-component.yaml [hunk 4]
```diff
@@ -17,4 +17,4 @@ kind: ComponentBuilder
 componentName: etcd-component
 version: 30.0.2
 objectFiles:
-- url: file:///etcd/etcd-server.yaml
+- url: ./etcd-server.yaml
```
[the remark this answers is not in the document]
Comment (Kashomon) [reply examples/cluster/etcd/etcd-component.yaml]: This uses url.Parse(...). Whatever that accepts is valid

File: pkg/build/path_rewriter.go (hunk 2 above)
Comment (Kashomon) [reply pkg/build/path_rewriter.go]: I thought that I might want different rewriters. /shrug. I don't have a strong opinion.

File: pkg/build/path_rewriter.go (hunk 3 above)
Comment (Kashomon) [reply pkg/build/path_rewriter.go]: Done.

Comment (Kashomon) [reply pkg/build/path_rewriter.go]: I don't know how to disallow traversals out of the workspace. =/

File: pkg/build/path_rewriter.go (hunk 1 above)
Comment (Kashomon) [reply pkg/build/path_rewriter.go]: I did MakeAbs(...)

**Comment by mikedanese** [comment]
LGTM

**Pull request merged**